src/main/drivers/gyro_sync.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * gyro_sync.c
+ *
+ *  Created on: 3 aug. 2015
+ *      Author: borisb
+ */
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "platform.h"
+#include "build_config.h"
+
+#include "common/axis.h"
+#include "common/maths.h"
+
+#include "sensor.h"
+#include "accgyro.h"
+
+#include "accgyro_mpu6050.h"
+#include "accgyro_spi_mpu6000.h"
+#include "sensors/sensors.h"
+#include "sensors/acceleration.h"
+#include "gyro_sync.h"
+
+#include "config/runtime_config.h"
+#include "config/config.h"
+
+bool gyroIsUpdated = false;
+
+uint32_t targetLooptime;
+static uint8_t mpuDivider;
+
+void setGyroRefreshRate(uint32_t looptime, uint8_t lpf) {
+#ifdef STM32F303xC
+    if (lpf == INV_FILTER_256HZ_NOLPF2) {
+        if (!sensors(SENSOR_ACC)) {
```

Comment:
The reason for this :D. Late night last change as I noticed that 2khz mode wasn't possible for my spracingF3 to keep up during my evening testing on different boards. It does keep up when acc disabled.